snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -76,6 +80,118 @@ def __init__(self, *, output, source, project_options,
             os.path.join('~', 'snap', 'lxd', 'common'))
         os.makedirs(self._lxd_common_dir, exist_ok=True)
 
+    def _get_multipass_remote(self):
+        try:
+            containers = subprocess.check_output([
+                'multipass', 'list']).decode()
+            # Trailing space to disambiguate names starting with snapcraft
+            if 'snapcraft ' not in containers:
+                subprocess.check_call([
+                    'multipass', 'launch', '--name=snapcraft', '--disk', '10G',
+                    ])
+            subprocess.check_call([
+                'multipass', 'start', 'snapcraft'])
+            self._setup_bridge()
+            self._setup_lxd()
+
+            return 'multipass'
+        except FileNotFoundError:
+            raise ContainerError('Multipass is not installed')
+        except subprocess.CalledProcessError as e:
+            raise ContainerError(
+                'Failed to setup multipass remote: {!r}'.format(
+                    ' '.join(e.cmd)))
+
+    def _setup_bridge(self):
+        logger.debug('Setting up the bridge')
+        # debconf database may be locked by another process, so loop and re-try
+        retry_count = 10
+        while True:
+            time.sleep(1)
+            try:
+                answers = dedent('''\
```

Comment:
this code will only work with lxd 2.0 and not later. So this by default with the snap.